opensrp-chw/src/main/java/org/smartregister/chw/interactor/ChildProfileInteractor.java

```diff
@@ -261,6 +274,60 @@ private void getDob(JSONObject jsonObject, String dobString) throws JSONExceptio
         }
     }
 
+    @Override
+    public void launchThinkMDHealthAssessment(@NotNull Context context) {
+        Runnable runnable = () -> {
+            try {
+                ChildFHIRBundleDao fhirBundleDao = new ChildFHIRBundleDao();
+                FHIRBundleModel bundle = fhirBundleDao.fetchFHIRDateModel(context, getChildBaseEntityId());
+                addThinkmdIdentifier(context, bundle.getUniqueIdGeneratedForThinkMD(), getChildBaseEntityId());
+                ThinkMDLibrary.getInstance().processHealthAssessment(context, bundle);
+            } catch (Exception e) {
+                Timber.e(e);
+            }
+        };
+
+        appExecutors.diskIO().execute(runnable);
+    }
+
+    private void addThinkmdIdentifier(@NotNull Context context, String uniqueIdGeneratedForThinkMD, @NotNull String childBaseEntityId) {
+        Event event = new Event()
+                .withBaseEntityId(childBaseEntityId)
+                .withEventType("update_thinkmd_id")
+                .withEntityType("ec_child")
+                .withEventDate(new Date())
+                .addIdentifier(context.getString(R.string.thinkmd_identifier_type), uniqueIdGeneratedForThinkMD);
+        event.withDateCreated(new Date());
+        tagSyncMetadata(ChwApplication.getInstance().getContext().allSharedPreferences(), event);
+
+        try {
+            // update event
+            JSONObject eventPartialJson = new JSONObject(JsonFormUtils.gson.toJson(event));
+            getSyncHelper().addEvent(childBaseEntityId, eventPartialJson);
+            // update local storage
+            AllCommonsRepository allCommonsRepository = CoreChwApplication.getInstance().getAllCommonsRepository("ec_child");
+            //Update REGISTER and FTS Tables
+            if (allCommonsRepository != null) {
+                ContentValues values = new ContentValues();
+                values.put(context.getString(R.string.thinkmd_identifier_type), uniqueIdGeneratedForThinkMD);
+                allCommonsRepository.update("ec_child", values, childBaseEntityId);
```

Comment:
instead of using this custom logic whey don't we use ec_client_fields to extract this value. That is better since it also works for events synced from server

Reply:
I am using both, ec_client_fields took some time to process the event and update the local DB, we immediately need to update the local DB so that, ThinkMD id immediately updated before launch the ThinkMD assessment.